web/src/stream_data.ts

```diff
@@ -1047,18 +1047,18 @@ export function create_sub_from_server_data(
         ...attrs,
     };
 
-    if (attrs.partial_subscribers !== undefined) {
-        peer_data.set_subscribers(sub.stream_id, attrs.partial_subscribers, false);
-    } else {
-        peer_data.set_subscribers(sub.stream_id, subscriber_user_ids ?? []);
-    }
-
     clean_up_description(sub);
 
     stream_info.set(sub.stream_id, sub);
     stream_ids_by_name.set(sub.name, sub.stream_id);
     sub_store.add_hydrated_sub(sub.stream_id, sub);
 
+    if (attrs.partial_subscribers !== undefined) {
```

Comment:
I think we want logic **before** we assign the big struct to `sub` with `...attrs);` at the end of it to deal with the possibility of having `partial_subscribers`.  The legacy code does ` delete attrs.subscribers;`, but I don't see where `delete attrs.partial_subscribers` happens.

Reply:
Something like:

~~~ js
full_data = attrs.partial_subscribers !== undefined;
subscriber_ids = full_data ? attr.subscribers : attr.partial_subscribers;
delete attrs.subscribers;
delete attrs.partial_subscribers;
sub = {
  // [...].
  ...attrs};

peer_data.set_subscribers(sub.stream_id, subscriber_user_ids, full_data);